tips/TIP-0053/tip-0053.md

```diff
@@ -0,0 +1,84 @@
+---
+tip: 53
+title: Tagged Data Payload
+description: Block payload for arbitrary data
+author: Philipp Gackstatter (@PhilippGackstatter) <philipp.gackstatter@iota.org>
+discussions-to: https://github.com/iotaledger/tips/pull/156
+status: Draft
+type: Standards
+layer: Core
+created: 2023-11-02
+replaces: 23
+---
+
+# Abstract
+
+This TIP is a slightly revised version of TIP-23 with updates to IOTA 2.0, like the the type prefix of the _Tagged Data_
+Payload type and its syntactic validation. The payload concept offers a very flexible way to combine and encapsulate
+information in the IOTA protocol. This document proposes a basic payload type that allows the addition of arbitrary
+data.
+
+# Motivation
+
+The most flexible way to extend an existing object is by the addition of arbitrary data. This payload provides a way to
+do just that. An optional tag can be used to categorize the data.
+
+# Specification
+
+## Tagged Data
+
+The following table describes the serialization of a _Tagged Data Payload_ following the notation from
+[TIP-21](../TIP-0021/tip-0021.md):
+
+<details>
+  <summary>Tagged Data</summary>
+  <blockquote>Data with an optional tag.</blockquote>
+</details>
+<table>
+  <tr>
+    <td>
+      <b>Name</b>
+    </td>
+    <td>
+      <b>Type</b>
+    </td>
+    <td>
+      <b>Description</b>
+    </td>
+  </tr>
+  <tr>
+    <td>Payload Type</td>
+    <td>uint8</td>
+    <td>Set to <strong>value 0</strong> to denote a <i>Tagged Data Payload</i>.</td>
+  </tr>
+  <tr>
+    <td>Tag</td>
+    <td>(uint8)ByteArray</td>
+    <td>The tag of the data.</td>
+  </tr>
+  <tr>
+    <td>Data</td>
+    <td>(uint32)ByteArray</td>
+    <td>Binary data.</td>
+  </tr>
+</table>
+
+It is important to note that `Tag` is not considered by the protocol, it just serves as a marker for second layer
+applications.
+
+## Syntactic Validation
+
+- The length of the `Tag` must not be larger than `64`.
+- The length of `Data` must not be larger than `8192`.
```

Comment:
Just to make sure, both tag and data can have length 0, right ? because the description says `Data with an optional tag` which makes it seem data can't be empty.

Reply:
I checked with the others and it's indeed optional. Updated the TIP to clarify.